dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -311,7 +303,7 @@ else if ( item.getEnrollment() != null )
         else if ( item.getEvent() != null )
         {
             result.setEvent(
-                eventService.getEvent( item.getEvent(),
+                eventService.getEvent( item.getEvent().getUid(),
```

Comment:
That is a tricky one!
It would be true if when you see an event you can know that it really is an event but here we are dealing with `detached` and `managed` entity by Hibernate.
If here I just get the event, sometimes (when it is coming as the side of a relationship), it is just a detached entity and it will not have a program that is required to check access.
We will need a more in depth discussion on entity management in the system